src/components/Settings/PhoneAuthentication.vue

```diff
@@ -120,39 +161,113 @@ export default {
 		return {
 			phoneNumber: '',
 			isPhoneNumberValid: false,
-			verificationType: '', // text or call
+			verificationType: '', // SMS or voice
 			userVerificationCode: '', // user entered number
+			oobCode: '',
+			isVerificationPending: false,
+			verificationError: '',
 			step: 0,
 		};
 	},
 	methods: {
+		getMFAToken() {
+			return new Promise((resolve, reject) => {
+				if (this.kvAuth0.enabled) {
+					this.kvAuth0.checkSession()
+						.then(() => this.kvAuth0.getMfaManagementToken())
+						.then(token => {
+							resolve(token);
+						}).catch(err => {
+							reject(err);
+						});
+				} else {
+					reject('Auth not enabled');
+				}
+			});
+		},
 		onValidityChanged(isValid) {
 			this.isPhoneNumberValid = isValid;
 		},
 		sendVerificationCode(verificationType) {
-			console.log('send');
-			this.userVerificationCode = ''; // clear out any existing codes.
 			this.verificationType = verificationType;
+			this.userVerificationCode = ''; // clear out any existing codes.
+			this.verificationError = '';
+			this.isVerificationPending = true;
 
-			// TODO: Hit endpoint to call or text based on verification method
-			// TODO: Setup some pending/error indication in the UI
+			const mutation = this.verificationType === 'SMS'
+				? enrollSMSAuthenticatorMutation
+				: enrollVoiceAuthenticatorMutation;
 
-			// if successful, move forward.
-			this.step = 1;
+			this.getMFAToken()
+				.then(token => {
+					return this.apollo.mutate({
+						mutation,
+						variables: {
+							mfa_token: token,
+							phone_number: this.phoneNumber
+						}
+					});
+				}).then(result => {
+					if (result.errors) {
+						throw result.errors;
+					} else {
+						this.oobCode = this.verificationType === 'SMS'
+							? result?.data?.my?.enrollSMSAuthenticator?.oob_code
+							: result?.data?.my?.enrollVoiceAuthenticator?.oob_code;
+						if (this.oobCode) {
+							this.step = 1;
+						}
+					}
+				})
+				.catch(err => {
+					console.error(err);
+					this.verificationError = err?.[0].message || err || 'Error. Please refresh the page and try again.';
```

Comment:
You can add the optional chaining to the array access as well to avoid "cannot read message of undefined" errors:
```
err?.[0]?.message
```